src/main/java/com/books/storage/concrete/SQL/AuthorSQLRepository.java

```diff
@@ -0,0 +1,137 @@
+package com.books.storage.concrete.SQL;
+
+import com.books.entities.Person;
+import com.books.storage.abstracts.Repository;
+import com.books.utils.AuthorTableColomnName;
+import com.books.utils.Constants;
+import org.h2.jdbcx.JdbcConnectionPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthorSQLRepository implements Repository<Person> {
+    private static final String AUTHOR_TABLE_NAME = "bookapp.authors";
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthorSQLRepository.class);
+    public static final AuthorSQLRepository INSTANCE = new AuthorSQLRepository();
+    private JdbcConnectionPool connectionPool;
+
+    public static AuthorSQLRepository getInstance() {
+        return INSTANCE;
+    }
+
+    private AuthorSQLRepository() {
+        connectionPool = JdbcConnectionPool.create(Constants.DATABASE_URL,
```

Comment:
Every repository has his own connection pool. This is a point for possible memory leaks.